framework/helpers/BaseConsole.php

```diff
@@ -855,21 +855,39 @@ public static function confirm($message, $default = false)
      *
      * @param string $prompt the prompt message
      * @param array $options Key-value array of options to choose from
+     * @param bool $displayList whether or not to initially display the option list
      *
      * @return string An option character the user chose
      */
-    public static function select($prompt, $options = [])
+    public static function select($prompt, $options = [], $displayList=false)
     {
+        $errorCount = 0;
+        if ($displayList) {
+            goto displayList;
```

Comment:
please avoid the use of goto here and create a new method for displaying the options.